### PR TITLE
Fetching the overridden GCP URL's from gcloud

### PR DESCRIPTION
**Pull request by harsha-accenture**

### CODE REVIEW

File: dataproc_jupyter_plugin/handlers.py [hunk 1]
```diff
@@ -119,6 +120,10 @@ def get_cached_credentials():
 
         return credentials
 
+def gcp_service_url(service_name, default_url=None):
+    default_url = default_url or f'https://{service_name}.googleapis.com/'
+    configured_url = get_gcloud_config(f'configuration.properties.api_endpoint_overrides.{service_name}')
+    return configured_url or default_url
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/handlers.py:126]: There's a missing blank line between the method definitions, but I may merge this and then fix it in a follow up.

**Review by ojarjur — APPROVED** [review]

File: dataproc_jupyter_plugin/handlers.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import datetime
 
 from google.cloud.jupyter_config.config import gcp_kernel_gateway_url
+from google.cloud.jupyter_config.config import get_gcloud_config
```
Comment (ptwng) [inline dataproc_jupyter_plugin/handlers.py:24]: Nit, should this (and the previous line) be:

`from google.cloud.jupyter_config.config import gcp_kernel_gateway_url, get_gcloud_config`

File: src/utils/const.ts [hunk 3]
```diff
@@ -15,15 +15,36 @@
...
+  compute_url: string;
+  metastore_url: string;
+  cloudkms_url: string;
+  cloudresourcemanger_url: string;
+  datacatalog_url: string;
+  storage_url: string;
+}
+
+const data = await requestAPI('getGcpServiceUrls');
+let gcpServiceUrl = data as IGcpUrlResponseData;
```
Comment (ptwng) [inline src/utils/const.ts:36]: nit: const

File: src/utils/const.ts [hunk 4]
```diff
@@ -142,11 +165,14 @@ export const SPARK_HISTORY_SERVER = 'Spark History Server';
 export const DEFAULT_LABEL_DETAIL = 'client:dataproc-jupyter-plugin';
 export const JOB_FIELDS_EXCLUDED = ['queryList', 'properties', 'args'];
 export const BATCH_FIELDS_EXCLUDED = ['queryList', 'properties'];
-export const GCS_URL = 'https://storage.googleapis.com/storage/v1/b';
-export const GCS_UPLOAD_URL = 'https://storage.googleapis.com/upload/storage/v1/b'
-export const KEY_MESSAGE='Example format:projects/<project-name>/locations/<location-name>/keyRings/<keyring-name>/cryptoKeys/<key-name>';
+export const GCS_URL = storedUrlResponseObj.storage_url + 'storage/v1/b';
```
Comment (ptwng) [inline src/utils/const.ts:168]: does this work?  It looks like in urlhandler we have:
`storage_url = gcp_service_url('storage', default_url='https://storage.googleapis.com/storage/v1/')`

Wouln't that mean that GCS_URL would be https://storage.googleapis.com/storage/v1/storage/v1/b

in the default case?

File: dataproc_jupyter_plugin/handlers.py [hunk 5]
```diff
@@ -183,6 +193,30 @@ def post(self):
...
+    @tornado.web.authenticated
+    def get(self):
+        url = {}
+        dataproc_url = gcp_service_url('dataproc')
+        compute_url = gcp_service_url('compute', default_url='https://compute.googleapis.com/compute/v1')
+        metastore_url = gcp_service_url('metastore')
+        cloudkms_url = gcp_service_url('cloudkms')
+        cloudresourcemanager_url = gcp_service_url('cloudresourcemanager')
+        datacatalog_url = gcp_service_url('datacatalog')
+        storage_url = gcp_service_url('storage', default_url='https://storage.googleapis.com/storage/v1/')
```
Comment (ptwng) [inline dataproc_jupyter_plugin/handlers.py:207]: I'm a bit confused as to when we have default urls with paths and when we don't.  Lets be consistant in how we do it across all endpoints?

File: src/utils/const.ts [hunk 6]
```diff
@@ -15,15 +15,36 @@
...
+  dataproc_url: string;
+  compute_url: string;
+  metastore_url: string;
+  cloudkms_url: string;
+  cloudresourcemanger_url: string;
+  datacatalog_url: string;
+  storage_url: string;
+}
+
+const data = await requestAPI('getGcpServiceUrls');
```
Comment (ptwng) [inline src/utils/const.ts:35]: Can we not do this in an await?  This is a blocking call while the javascript code initializes.

**Review by ptwng — CHANGES_REQUESTED** [review]

File: dataproc_jupyter_plugin/handlers.py (hunk 5 above)
Comment (ojarjur) [reply dataproc_jupyter_plugin/handlers.py:207]: This comes from `gcloud`... the defaults it uses differs depending on the API.

File: src/utils/const.ts (hunk 6 above)
Comment (harsha-accenture) [reply src/utils/const.ts:35]: @ptwng we tried removing that, but looks like we are not able to achieve what we are looking for. Please let us know if we need to try something else. The const.ts file is just static file with no react components. We do not see any lag in the UI as of now.

File: dataproc_jupyter_plugin/handlers.py [hunk 7]
```diff
@@ -203,7 +203,7 @@ def get(self):
         cloudkms_url = gcp_service_url('cloudkms')
         cloudresourcemanager_url = gcp_service_url('cloudresourcemanager')
         datacatalog_url = gcp_service_url('datacatalog')
-        storage_url = gcp_service_url('storage', default_url='https://storage.googleapis.com/storage/v1/')
+        storage_url = gcp_service_url('storage', default_url='https://content-storage.googleapis.com')
```
Comment (ojarjur) [inline dataproc_jupyter_plugin/handlers.py:206]: This should change back to the previous version. In particular, the value returned for the `api_endpoint_overrides/storage` setting should always include the `/storage/<VERSION/` path.

**Review by ptwng — APPROVED** [review]

**Pull request merged**